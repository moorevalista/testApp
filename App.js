/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import type {Node} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import {NavigationContainer} from '@react-navigation/native';
import Router from './src/router';

import {
  Colors,
} from 'react-native/Libraries/NewAppScreen';

const App: () => Node = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : "rgba(248,248,255,1)",
  };

  return (
    <NavigationContainer>
      <SafeAreaView style={[styles.container, {backgroundStyle}]}>
        <StatusBar
          barStyle={isDarkMode ? 'light-content' : 'dark-content'}
          backgroundColor={backgroundStyle.backgroundColor}
        />
        
          <View
            style={[styles.container ,{
              backgroundColor: isDarkMode ? Colors.black : Colors.white,
            }]}>
            <Router />
          </View>
      </SafeAreaView>
    </NavigationContainer>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "rgba(248,248,255,1)",
  },
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default App;
