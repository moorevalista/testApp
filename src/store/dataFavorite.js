// const initialState = {
//   dataFavorite: []
// }

const dataFavorite = (state = [], action) => {
  // console.log('Payload : ', action.payload);
  switch(action.type) {
    case 'setData':
      return { 
        state: action.payload
      }
      break;
    // case 'addItem':
    //   return { 
    //     ...state,
    //     dataFavorite: [...state.dataFavorite, action.payload]
    //   }
    //   break;
    default:
      return state;
  }
  return state;
};
  
export default dataFavorite;  