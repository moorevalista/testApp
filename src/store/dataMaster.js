
const dataMaster = (state = '', action) => {
  // console.log('Payload : ', action.payload);
  switch(action.type) {
    case 'setDataMaster':
      return { 
        state: action.payload
      }
      break;
    default:
      return state;
  }
  return state;
};
  
export default dataMaster;  