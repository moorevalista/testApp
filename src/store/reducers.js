import {combineReducers} from 'redux';
import dataFavorite from './dataFavorite';
import dataMaster from './dataMaster';

const allReducers = combineReducers({
  dataFavorite,
  dataMaster,
});

export default allReducers;
