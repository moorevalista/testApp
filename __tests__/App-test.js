import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import { Provider } from 'react-redux';
import store from '../src/store';
// import configureStore from 'redux-mock-store';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

configure({adapter: new Adapter(), disableLifecycleMethods: true})
// const initialState = [1,2,3];
// const store = configureStore(initialState)
const appWrapper = shallow(<Provider store={store}><App /></Provider>)

describe('App Screen', () => {
  
  it('should renders correctly', () => {
    expect(renderer.create(<Provider store={store}><App /></Provider>).toJSON()).toMatchSnapshot();
  })

  it('should renders `App Screen` module correctly', () => {
    expect(appWrapper).toMatchSnapshot()
  })

})