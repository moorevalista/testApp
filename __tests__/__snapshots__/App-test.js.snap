// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Screen should renders \`App Screen\` module correctly 1`] = `ShallowWrapper {}`;

exports[`App Screen should renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "backgroundColor": "rgba(248,248,255,1)",
        "flex": 1,
      },
      Object {
        "backgroundStyle": Object {
          "backgroundColor": "rgba(248,248,255,1)",
        },
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(248,248,255,1)",
          "flex": 1,
        },
        Object {
          "backgroundColor": "#FFF",
        },
      ]
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          stackPresentation="push"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="owner"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </RNCSafeAreaProvider>
  </View>
</RCTSafeAreaView>
`;
